pact/verifier.py

```diff
@@ -28,68 +28,160 @@ def __str__(self):
         return 'Verifier for {} with url {}'.format(self.provider, self.provider_base_url)
 
     def validate_publish(self, **kwargs):
-        """Validate publish has a version."""
-        if (kwargs.get('publish') is not None) and (kwargs.get('publish_version') is None):
-            # do something
-            raise Exception()
-
-    def verify_pacts(self, *pacts, enable_pending=False, include_wip_pacts_since=None, **kwargs):
-        """Verify our pacts from the provider.
-
-        Returns:
-          success: True if no failures
-          logs: some tbd output of logs
+        """Ensure the participant version is provided.
 
+        :raises Exception: When `publish_verification_results` is set,
+            but `publish_version` is not provided.
+        """
+        if not kwargs.get('publish_verification_results'):
+            return
+
+        if not kwargs.get('publish_version'):
+            raise Exception('To publish the results of verification to the '
+                            'broker version of the participant is mandatory')
+
+    def verify_pacts(
+            self,
+            *pacts,
+            enable_pending=False,
+            include_wip_pacts_since=None,
+            **kwargs
+    ):
+        """Verify pacts from the provider.
+
+        Usage:
+
+        >>> from pact import Verifier
+        >>>
+        >>> verifier = Verifier(
+        ...   provider='UserService',
+        ...   provider_base_url='http://localhost:5001'
+        ... )
+        >>>
+        >>> result, _ = verifier.verify_pacts(
+        ...   'consumer-provider.json',  # Optional
+        ...   enable_pending=True,
+        ... )
+        >>>
+        >>> assert result == 0
+
+        :param pacts: Optional list of pacts to verify. Every pact in that list
```

Comment:
I don't think it is actually optional, because if you have no pacts defined here, and there is no broker (if using with a broker you would call verify_with_broker instead I think) it will fail in `verify_wrapper._validate_input`

Reply:
No it isn't. Right now you can call this function without this parameter and it will work correctly. Besides, my edits didn't change anything in this regard

Reply:
What do you mean will work correctly?
If I call it without anything for pacts:
```
verifier = Verifier(
           provider='UserService',
           provider_base_url='http://localhost:5001'
)
result, _ = verifier.verify_pacts()
```

It will error:
```
    def _validate_input(self, pacts, **kwargs):
        if len(pacts) == 0 and not self._broker_present(**kwargs):
>           raise PactException('Pact urls or Pact broker required')
E           pact.verify_wrapper.PactException: Pact urls or Pact broker required

../pact/verify_wrapper.py:135: PactException
```

There must be at least one pact provided, it isn't `Optional`. I am talking about the description, it shouldn't be described as optional when it will not work without...

Reply:
I see. I corrected it. Thank you for your patience.